### PR TITLE
Improve accuracy and clarity of JavaScript/TypeScript server docs

### PR DESCRIPTION
**Pull request by aspiers**

### CODE REVIEW

**Review by ericdallo — APPROVED** [review]

**Comment by aspiers** [comment]
Looks like a test failed due to a glitch with MELPA.  I pushed another commit anyway, so hopefully that will retrigger everything.

File: clients/lsp-javascript.el [hunk 1]
```diff
@@ -60,8 +61,10 @@
                                         (lsp-package-ensure
                                          'javascript-typescript-langserver
                                          callback
-                                         error-callback))))
-
+                                         error-callback))
+                  :initialized-fn (lambda (workspace)
```
Comment (yyoncho) [inline clients/lsp-javascript.el:65]: make this _workspace to make CI happy.

Comment (aspiers) [reply clients/lsp-javascript.el]: Done

**Review by jcs090218 — APPROVED** [review]

**Comment by yyoncho** [comment]
Thank you! Merging as it is. We may reconsider the warning in case a lot of people complain.

**Pull request merged**